bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/client/Client.java

```diff
@@ -91,12 +88,12 @@ public class Client {
 
         @SuppressWarnings("rawtypes")
         @Override
-        public @Nullable Iterator getPrefixes(@Nullable String val) {
+        public @Nullable Iterator getPrefixes(@Nullable String namespaceURI) {
```

Comment:
I had a hard time here, but the warning remains:
```
[WARNING] Client.java:[91,26] Type safety: The return type java.util.Iterator for getPrefixes(java.lang.String) from the type new javax.xml.namespace.NamespaceContext(){} needs unchecked conversion to conform to java.util.Iterator<java.lang.String> from the type javax.xml.namespace.NamespaceContext
```

First step was to change to correct method signature:
```java
public @Nullable Iterator<String> getPrefixes(@Nullable String namespaceURI) {
```

But now having null annotation issues. I tried all combinations of `@Nullable`/`@NonNull`, but none of them were accepted.

Reply:
I have faced similar challanges and finally gave up. I think it is acceptable to have a few warnings like this. We have like 10k other warnings to fix ;-) Escpecially when null annotations are added to all bindings. Must say we do make progress on this subject, slowly, but progress!